runtime/primitives/src/assets.rs

```diff
@@ -134,3 +136,97 @@ where
 			.ok_or(())
 	}
 }
+
```

Comment:
If this enum will be used from frontend, can we derive some necessary traits here?

Reply:
I don't think front-end will use this.